app/bundles/DynamicContentBundle/Helper/DynamicContentHelper.php

```diff
@@ -118,4 +137,110 @@ public function replaceTokensInContent($content, $lead)
 
         return $content;
     }
+
+    /**
+     * @param string    $content
+     * @param Lead|null $lead
+     *
+     * @return array
+     */
+    public function findDwcTokens($content, $lead)
+    {
+        preg_match_all('/{dwc=(.*?)}/', $content, $matches);
+
+        $tokens = [];
+        if (!empty($matches[1])) {
+            foreach ($matches[1] as $key => $slotName) {
+                $token = $matches[0][$key];
+                if (!empty($tokens[$token])) {
+                    continue;
+                }
+
+                $dwcs = $this->getDwcsBySlotName($slotName);
+
+                /** @var DynamicContent $dwc */
+                foreach ($dwcs as $dwc) {
+                    if ($dwc->getIsCampaignBased()) {
+                        continue;
+                    }
+                    $content                   = $lead ? $this->getRealDynamicContent($dwc->getSlotName(), $lead, $dwc) : '';
+                    $tokens[$token]['content'] = $content;
+                    $tokens[$token]['filters'] = $dwc->getFilters();
+                }
+            }
+
+            unset($matches);
+        }
+
+        return $tokens;
+    }
+
+    /**
+     * @param $slot
+     * @param $lead
+     * @param $dwc
+     *
+     * @return string
+     */
+    public function getRealDynamicContent($slot, $lead, DynamicContent $dwc)
```

Comment:
Could we typehint parameters? What is `$slot`? Is it possible to use `Lead $lead`?